tests/src/App/ComplexGeoData.cpp

```diff
@@ -0,0 +1,427 @@
+    auto indexedName = Data::IndexedName("EDGE1");
+
+    // Act
+    auto result = cgd().getElementMappedNames(indexedName, true);
+
+    // Assert
+    EXPECT_FALSE(result.empty());
+}
+
+
+TEST_F(ComplexGeoDataTest, getElementMappedNamesWithMapWithUnmapped)// NOLINT
+{
+    // Arrange
+    Data::MappedName mappedName;
+    Data::IndexedName indexedName;
+    std::tie(indexedName, mappedName) = createMappedName("TestMappedName");
+    auto unmappedName = Data::IndexedName("EDGE", 2);
+
+    // Act
+    auto result = cgd().getElementMappedNames(unmappedName, true);
+
+    // Assert
+    EXPECT_FALSE(result.empty());
+    EXPECT_NE(result[0].first, mappedName);
+}
+
+TEST_F(ComplexGeoDataTest, elementTypeValidIndexName)// NOLINT
+{
+    // Arrange
+    auto indexedName = Data::IndexedName("EDGE", 1);
+
+    // Act
+    char elementType = cgd().elementType(indexedName);
+
+    // Assert
+    EXPECT_EQ(elementType, 'E');
+}
+
+TEST_F(ComplexGeoDataTest, elementTypeInvalidIndexedName)// NOLINT
+{
+    // Arrange
+    auto indexedName = Data::IndexedName("INVALID", 1);// Not in the element type list
+
+    // Act
+    char elementType = cgd().elementType(indexedName);
+
+    // Assert
+    EXPECT_EQ(elementType, 0);
+}
+
+TEST_F(ComplexGeoDataTest, elementTypeCharEmptyName)// NOLINT
+{
+    // Arrange & Act
+    char elementType = cgd().elementType(nullptr);
+
+    // Assert
+    EXPECT_EQ(elementType, 0);
+}
+
+TEST_F(ComplexGeoDataTest, elementTypeCharIndexedName)// NOLINT
+{
+    // Arrange & Act
+    char elementType = cgd().elementType("EDGE1");
+
+    // Assert
+    EXPECT_EQ(elementType, 'E');
+}
+
+TEST_F(ComplexGeoDataTest, elementTypeCharMappedNameNoPrefix)// NOLINT
+{
+    // Arrange
+    int size {0};
+    Data::MappedName mappedName;
+    Data::IndexedName indexedName;
+    std::tie(indexedName, mappedName) = createMappedName("TestMappedName:;");
+
+    // Act
+    char elementType = cgd().elementType(mappedName.toConstString(0, size));
+
+    // Assert
+    EXPECT_EQ(elementType, 'E');
+}
+
+TEST_F(ComplexGeoDataTest, elementTypeCharMappedNameWithPrefix)// NOLINT
+{
+    // Arrange
+    int size {0};
+    Data::MappedName mappedName;
+    Data::IndexedName indexedName;
+    auto name = fmt::format("{}TestMappedElement:;", Data::ELEMENT_MAP_PREFIX);
+    std::tie(indexedName, mappedName) = createMappedName(name);
+
+    // Act
+    char elementType = cgd().elementType(mappedName.toConstString(0, size));
+
+    // Assert
+    EXPECT_EQ(elementType, 'E');
+}
+
+TEST_F(ComplexGeoDataTest, resetElementMapNoArgument)// NOLINT
+{
+    // Arrange & Act
+    cgd().resetElementMap();
+
+    // Assert
+    EXPECT_EQ(cgd().getElementMapSize(), 0);
+}
+
+TEST_F(ComplexGeoDataTest, resetElementMapWithArgument)// NOLINT
+{
+    // Arrange
+    auto elementMap = std::make_shared<Data::ElementMap>();
+    auto mappedName = Data::MappedName(";TestMappedName");
+    auto indexedName = Data::IndexedName(cgd().getElementTypes().front(), 1);
```

Comment:
if mappedname and indexedname are unimportant (from what I understand they're just to get size 1) I would use more generic names for the sake of simplicity and readability.
- in mappedname that leading ';' might seem an important detail
- in indexedname `cgd().getElementTypes().front()` might suggest some kind of coupling to complexgeodata

Same for a couple of tests below

Reply:
Good suggestions, I will change.